File: leetcode/medium/2044. Count Number of Maximum Bitwise-OR Subsets/main.go
package main

func countMaxOrSubsets(nums []int) int {
	var (
		result = 0
		maxOr  = 0
	)
	for _, num := range nums {
		maxOr |= num
	}

	backtrack(nums, 0, 0, maxOr, &result)

	return result
}

func backtrack(nums []int, idx, currOr, maxOr int, result *int) {
	if currOr == maxOr {
		*result += 1
	}

	for i := idx; i < len(nums); i++ {
		backtrack(nums, i+1, currOr|nums[i], maxOr, result)
	}
}

File: leetcode/medium/2044. Count Number of Maximum Bitwise-OR Subsets/main.ts
const countMaxOrSubsets = (nums: number[]): number => {
  let [maxOr, result] = [0, 0];

  for (let num of nums) {
    maxOr |= num;
  }

  const backtrack = (idx: number, currOr: number) => {
    if (currOr == maxOr) {
      result += 1;
    }

    for (let i = idx; i < nums.length; i++) {
      backtrack(i + 1, currOr | nums[i]);
    }
  }

  backtrack(0, 0);

  return result
};
